### ISSUE TITLE
`device_args` responsive conditions not working correctly for `choose` and `dimension` controls since v3.4

### ISSUE THREAD

**Issue by parasshah195** [elementor/elementor#16292]
Hello team,

I have an addon which adds a custom `dimension` control responsive conditioned on the value of a `choose` control and it's not behaving correctly since v3.4.

I've been in contact with @udidol about this since v3.4-beta and he helped me improve my previous implementation to make it more performant and optimal, but the bug still continues to happen.

### Here's what exactly happens

**Expected behavior:**

On changing the value of the `choose` control on any breakpoint, the relevant `dimension` control should be shown

**Current glitchy behavior:**

When a **non-default value** of the `choose` control is selected on any breakpoint (except desktop), changing its value on smaller breakpoint doesn't recognize or process the condition until an up/down arrow key is pressed in the visible `dimension` control.

I have recorded a short 1-min screencast of the issue in action here - https://www.loom.com/share/5971b9cd28e14c108c015f351da0c32e

It works fine on previous versions before 3.4.

### Sample code similar to my implementation

```
add_action( 'elementor/element/column/layout/after_section_start', function( $control_stack ) {
 
    $control_stack->add_responsive_control(
        'custom_col_position_type',
        [
            'label' => __( 'Positioning', 'plugin-name' ),
            'type' => \Elementor\Controls_Manager::CHOOSE,
            'options' => [
                'auto' => [
                    'title' => __( 'Auto', 'plugin-name' ),
                    'icon' => 'fa fa-th-large',
                ],
                'manual' => [
                    'title' => __( 'Manual', 'plugin-name' ),
                    'icon' => 'fa fa-edit',
                ],
            ],
            'default' => 'auto',
            'style_transfer' => false,
            'toggle' => 'false',
            'render_type' => 'ui',
        ]
    );
 
    /**
     * If version greater than 3.4.0, use the new breakpoints class
     */
    if ( version_compare( ELEMENTOR_VERSION, '3.4.0', '>=' ) ) {
        $breakpoints = \Elementor\Plugin::$instance->breakpoints->get_active_devices_list();
    } else {
        // previous versions
        $breakpoints = [ 'desktop', 'tablet', 'mobile' ];
    }
 
    custom_positioning_controls( $control_stack, $breakpoints );
 
});
 
function custom_positioning_controls( $control_stack, $breakpoints ) {
 
    $auto_control_device_args = custom_positioning_generate_device_args( $breakpoints, 'auto' );
    $manual_control_device_args = custom_positioning_generate_device_args( $breakpoints, 'manual' );
 
    $control_stack->add_responsive_control(
        'custom_col_position_auto',
        [
            'label'      => __( 'Auto', 'designer-powerup-for-elementor' ),
            'type' => \Elementor\Controls_Manager::DIMENSIONS,
            'size_units' => [],
            'style_transfer' => false,
            'selectors' => [
                '{{WRAPPER}}' => 'background-color: yellow', //temporary yellow color bg CSS for auto position
            ],
            'device_args' => $auto_control_device_args,
        ]
    );
 
    $control_stack->add_responsive_control(
        'custom_col_position_manual',
        [
            'label'      => __( 'Manual', 'designer-powerup-for-elementor' ),
            'type' => \Elementor\Controls_Manager::DIMENSIONS,
            'size_units' => [],
            'style_transfer' => false,
            'selectors' => [
                '{{WRAPPER}}' => 'background-color: red', //temporary red color bg CSS for manual position
            ],
            'device_args' => $manual_control_device_args,
        ]
    );
}
 
/**
 * Generate `device_args` key with condition for positioning type control
 *
 * @param array $breakpoints A list of enabled breakpoint names
 * @param string $value auto|manual
 */
function custom_positioning_generate_device_args( $breakpoints, $value = 'auto' ) {
    $device_args = [];
 
    // populate device args with all active breakpoint conditions
    foreach( $breakpoints as $breakpoint ) {
 
        $control_id = ( 'desktop' === $breakpoint ) ? 'custom_col_position_type' : "custom_col_position_type_{$breakpoint}";
 
        $device_args[ $breakpoint ] = [
            'condition' => [
                $control_id => $value
            ]
        ];
    }
 
    return $device_args;
}
```
 
Hoping that this will be fixed soon as the addon users are facing this issue on their production sites which is breaking designs.

**Comment by udidol** [comment]
A fix for this issue will be released in the upcoming 3.5.0 **Beta 2** version.

**Comment by parasshah195** [comment]
Just an update from the internal conversation with @udidol, this wasn't entirely fixed in the 3.5.0-beta-2, but rather, it fully broke the functionality. Since then, the control does not output any responsive CSS on the frontend at all. This is breaking websites using the addon.

**Comment by udidol** [comment]
We have been working on the responsive control system in the past couple of weeks. We will be releasing fixes for responsive conditions issues in the upcoming Elementor 3.6.0 version, or even in a patch version to come before it.
We will update when a potential fix for your problem is released.

**Comment by parasshah195** [comment]
@udidol This now seems to be fixed in v3.6.0

Thank you :)

**Comment by udidol** [comment]
Amazing @parasshah195
